### PR TITLE
UI Initial Model Management Changes

### PR DESCRIPTION
**Pull request by estohlmann**
Extensive updates to add Model Management changes to the UI. Updates include:

- Adding State
- Adding UI support for initial Model Management Screen
- Adding API Support for Models (List/Start/Stop/Delete). First time we pulled in Axios and not used the Fetch API due to required HTTP Verbs
- Adding Notification Platform
- Updating Typescript Linting Rules to actually run some checks

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/api-base/fastApiContainer.ts [hunk 1]
```diff
@@ -35,97 +35,97 @@ const CONTAINER_MEMORY_BUFFER = 1024 * 2;
...
+                PYPI_TRUSTED_HOST: config.pypiConfig.trustedHost,
+                LITELLM_CONFIG: yamlDump(config.litellmConfig),
+            };
+        }
+        const environment: Record<string, string> = {
+            LOG_LEVEL: config.logLevel,
+            AWS_REGION: config.region,
+            AWS_REGION_NAME: config.region, // for supporting SageMaker endpoints in LiteLLM
+            THREADS: Ec2Metadata.get(taskConfig.instanceType).vCpus.toString(),
+            LITELLM_KEY: config.litellmConfig.general_settings.master_key, //
```
Comment (jtblack-aws) [inline lib/api-base/fastApiContainer.ts:84]: nit: can remove trailing `//`

File: lib/user-interface/react/src/components/model-management/ModelManagementActions.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        onClick={() => dispatch(modelManagementApi.util.invalidateTags(['models']))}
+        ariaLabel={'Refresh models cards'}
+      >
+        <Icon name='refresh' />
+      </Button>
+    </SpaceBetween>
+  );
+}
+
+function ModelActionButton(notificationService: INotificationService, props?: any) {
```
Comment (jtblack-aws) [inline lib/user-interface/react/src/components/model-management/ModelManagementActions.tsx:61]: return type

File: lib/api-base/fastApiContainer.ts (hunk 1 above)
Comment (petermuller) [reply lib/api-base/fastApiContainer.ts:84]: +1 or we can keep the comment. either way is fine for me.

File: lib/user-interface/react/src/pages/ModelManagement.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import { useEffect } from 'react';
+import ModelManagementComponent from '../components/model-management/ModelManagementComponent';
+
+export function ModelManagement ({ setTools }) {
```
Comment (jtblack-aws) [inline lib/user-interface/react/src/pages/ModelManagement.tsx:20]: return type

File: lib/user-interface/react/src/shared/notification/notification.service.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            generateNotification(`Failed to ${action}: ${actionResult.payload}`, 'error');
+        }
+    }
+
+    return {
+        generateNotification: generateNotification,
+        createNotification: createNotification,
+        showActionNotification: showActionNotification,
+    };
+}
```
Comment (jtblack-aws) [inline lib/user-interface/react/src/shared/notification/notification.service.tsx:58]: lots of missing return types

File: lib/user-interface/react/src/shared/notification/notification.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import { Flashbar } from '@cloudscape-design/components';
+import React, { useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { reset, selectNotifications } from '../reducers/notification.reducer';
+import { NotificationProp } from './notifications.props';
+import { useNotificationService } from '../util/hooks';
+import { useAppSelector } from '../../config/store';
+
+function NotificationBanner () {
```
Comment (jtblack-aws) [inline lib/user-interface/react/src/shared/notification/notification.tsx:25]: return type

File: lib/user-interface/react/src/shared/notification/notification.service.tsx (hunk 4 above)
Comment (petermuller) [reply lib/user-interface/react/src/shared/notification/notification.service.tsx:58]: is there another linting option we can set for these to catch them all?

**Review by jtblack-aws — DISMISSED** [review]
We can add return types to functions is subsequent PRs so not going to block on fixing them all now.

File: lib/user-interface/react/src/shared/notification/notification.service.tsx (hunk 4 above)
Comment (estohlmann) [reply lib/user-interface/react/src/shared/notification/notification.service.tsx:58]: Good question, I will look into a linter for this in a follow up.

**Review by jtblack-aws — APPROVED** [review]

**Pull request merged**